ex03.exs

```diff
@@ -49,13 +49,29 @@ defmodule Ex03 do
      it contains no conditional logic:  3
      it is nicely structured            7
   """
-
+ 
   def pmap(collection, process_count, function) do
-    # your code goes here
-    # I'm hoping to see a simple pipeline as the body of this function...
+    Enum.chunk_every(collection, trunc(Enum.count(collection)/process_count))
+     |>  Ex03.mapping(function)
+     |>  List.flatten()
   end
 
-  # and here...
+  def mapping(comchunk,function) do
+  initial = self()                  # else it will cause a timeout 
+    Enum.map(comchunk, fn (chunk) ->
+      spawn(fn ->   
+        send initial, {self(), Enum.map(chunk, function)}
+       end
+      ) 
+     end
+    )    
+   |> Enum.map(fn (pid) ->
+      receive do
+        {^pid, variable} -> variable 
+           end
+        end
+      )
+    end
```

Comment:
Stylistically, I have two comments. 

First, the `mapping` function is really long, and it does two distinct things. My feeling is that it should be split into at least two, so the top level pmap would become:

~~~ elixir
collection
|> Enum.chunk_every(trunc(Enum.count(collection)/process_count))
|>  run_in_parallel(function)
|> gather_results()
~~~

Also, the `List.flatten` wouldn't be needed if the `map` on line 68 was `flat_map`.

# -2
     |>  List.flatten()